### PR TITLE
Support for react-native@0.73

### PR DESCRIPTION
**Pull request by piaskowyk**
## Summary

The PR adds support for react-native@0.73.

#### Paper

```
* What went wrong:
A problem occurred configuring project ':react-native-reanimated'.
> defaultConfig contains custom BuildConfig fields, but the feature is disabled.
  To enable the feature, add the following to your module-level build.gradle:
  `android.buildFeatures.buildConfig true`
```
According to documentation, we can do it safely - https://developer.android.com/reference/tools/gradle-api/7.0/com/android/build/api/dsl/BuildFeatures#buildConfig:kotlin.Boolean

- https://github.com/facebook/react-native/pull/40748

#### Fabric

- https://github.com/facebook/react-native/pull/37588/files#diff-1aaade82ec15d197d2785561ee85f90029c7ebb46e75bc4a24967da1fc1cc041
- https://github.com/facebook/react-native/pull/36609/files#diff-e8a960fbc5a09cea2611fa92b7791c749c3fce1d02b614694d203d9712e97aa4
- https://github.com/facebook/react-native/pull/38276/files#diff-f53e9cf2e2c954166cb7a3cab84622f14af9eb3949dc1f29f76e4693a9e9cbde
- https://github.com/facebook/react-native/pull/36395/files#diff-2b25a23f0ed5ac2cf193ff05df77004c3eae39d5966e86a06d383f707d364cc8
- https://github.com/facebook/react-native/commit/33ebb5d6365c3dd5d5214afc42c373d265280e27#diff-efac1ff476300ebbe947340c1864ad44dae2ae4f24ab7837395b337303db6313

### Backward compatibility
image

### CODE REVIEW

File: Common/cpp/NativeModules/NativeReanimatedModule.cpp [hunk 1]
```diff
@@ -4,6 +4,9 @@
 #include <react/renderer/core/TraitCast.h>
 #include <react/renderer/uimanager/UIManagerBinding.h>
 #include <react/renderer/uimanager/primitives.h>
+#if REACT_NATIVE_MINOR_VERSION >= 73
```
Comment (tomekzaw) [inline Common/cpp/NativeModules/NativeReanimatedModule.cpp:7]:
```suggestion
#if REACT_NATIVE_MINOR_VERSION >= 73 && defined(RCT_NEW_ARCH_ENABLED)
```

File: Common/cpp/NativeModules/NativeReanimatedModule.cpp [hunk 2]
```diff
@@ -522,16 +529,20 @@ bool NativeReanimatedModule::handleRawEvent(
     // just ignore this event, because it's an event on unmounted component
     return false;
   }
-  const std::string &type = rawEvent.type;
-  const ValueFactory &payloadFactory = rawEvent.payloadFactory;
 
   int tag = eventTarget->getTag();
-  std::string eventType = type;
+  std::string eventType = rawEvent.type;
```
Comment (tomekzaw) [inline Common/cpp/NativeModules/NativeReanimatedModule.cpp:534]:
```suggestion
  const auto &eventType = rawEvent.type;
```

File: Common/cpp/NativeModules/NativeReanimatedModule.cpp [hunk 3]
```diff
@@ -30,6 +33,10 @@
 
 using namespace facebook;
 
+#if REACT_NATIVE_MINOR_VERSION >= 73 && RCT_NEW_ARCH_ENABLED
```
Comment (tomekzaw) [inline Common/cpp/NativeModules/NativeReanimatedModule.cpp:36]:
```suggestion
#if REACT_NATIVE_MINOR_VERSION >= 73 && defined(RCT_NEW_ARCH_ENABLED)
```

**Comment by lafest** [comment]
I met below error in RN0.73-rc.2. is there any change with handling hermes?
if I comment out enableDebuggging and disableDebugging, build successfully completed.

```
Undefined symbols for architecture x86_64:
  "facebook::hermes::inspector_modern::chrome::enableDebugging(std::__1::unique_ptr<facebook::hermes::inspector_modern::RuntimeAdapter, std::__1::default_delete<facebook::hermes::inspector_modern::RuntimeAdapter>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)", referenced from:
      reanimated::ReanimatedHermesRuntime::ReanimatedHermesRuntime(std::__1::unique_ptr<facebook::hermes::HermesRuntime, std::__1::default_delete<facebook::hermes::HermesRuntime>>, std::__1::shared_ptr<facebook::react::MessageQueueThread> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>) in libRNReanimated.a(ReanimatedHermesRuntime.o)
  "facebook::hermes::inspector_modern::chrome::disableDebugging(int)", referenced from:
      reanimated::ReanimatedHermesRuntime::~ReanimatedHermesRuntime() in libRNReanimated.a(ReanimatedHermesRuntime.o)
ld: symbol(s) not found for architecture x86_64
```

**Comment by gabrieldonadel** [comment]
> ⚠️ PR is ready to review/merge but let't wait for next rn-cli release, because now half of our CIs are falling due to: [react-native-community/cli#2096](https://github.com/react-native-community/cli/pull/2096)

Btw, seems that https://github.com/react-native-community/cli/pull/2096 was released in https://github.com/react-native-community/cli/releases/tag/v11.3.9

**Comment by matinzd** [comment]
Are you releasing this to nightly tag?

File: Common/cpp/NativeModules/NativeReanimatedModule.cpp [hunk 4]
```diff
@@ -416,16 +420,20 @@ bool NativeReanimatedModule::handleRawEvent(
...
     eventType = "on" + eventType.substr(3);
   }
   jsi::Runtime &rt = uiWorkletRuntime_->getJSIRuntime();
+#if REACT_NATIVE_MINOR_VERSION >= 73
+  const SharedEventPayload &eventPayload = rawEvent.eventPayload;
+  jsi::Value payload = eventPayload->asJSIValue(rt);
+#else
+  const ValueFactory &payloadFactory = rawEvent.payloadFactory;
   jsi::Value payload = payloadFactory(rt);
+#endif
```
Comment (tomekzaw) [inline Common/cpp/NativeModules/NativeReanimatedModule.cpp:436]:
```suggestion
#if REACT_NATIVE_MINOR_VERSION >= 73
  const auto &eventPayload = rawEvent.eventPayload;
  jsi::Value payload = eventPayload->asJSIValue(rt);
#else
  const auto &payloadFactory = rawEvent.payloadFactory;
  jsi::Value payload = payloadFactory(rt);
#endif
```

File: RNReanimated.podspec [hunk 5]
```diff
@@ -30,7 +30,7 @@ Pod::Spec.new do |s|
   s.license      = "MIT"
   # s.license    = { :type => "MIT", :file => "FILE_LICENSE" }
   s.author       = { "author" => "author@domain.cn" }
-  s.platforms    = { :ios => "9.0", :tvos => "9.0", :osx => "10.14" }
+  s.platforms    = { :ios => "10.0", :tvos => "9.0", :osx => "10.14" }
```
Comment (tomekzaw) [inline RNReanimated.podspec:33]: Can we set `:ios` based on `react_native_minor_version`? This way we can preserve compatibility with iOS 9.0 on older RN versions.

**Review by tomekzaw — APPROVED** [review]
👏

**Comment by efstathiosntonas** [comment]
Hey guys, please do not forget about this one: https://github.com/software-mansion/react-native-reanimated/pull/5204

**Comment by tomekzaw** [comment]
@efstathiosntonas Thanks for the reminder! Looks like this PR already contains these changes:

Zrzut ekranu 2023-10-30 o 10 06 46

File: RNReanimated.podspec (hunk 5 above)
Comment (piaskowyk) [reply RNReanimated.podspec]: sure thing 🫡 done in https://github.com/software-mansion/react-native-reanimated/pull/4913/commits/bb531b3db12f1d2b7abc2d5b12dafef836bfd696

**Review by tomekzaw — APPROVED** [review]
:shipit:

**Comment by lafest** [comment]
@piaskowyk Hi, can you check this comment? https://github.com/software-mansion/react-native-reanimated/pull/4913#issuecomment-1770033369

**Comment by piaskowyk** [comment]
Hey, could you tell me how I can reproduce this error?

**Comment by lafest** [comment]
@piaskowyk
I used xcode 14.3, react-native@0.73.0-rc.3, CLANG_CXX_LANGUAGE_STANDARD = "gnu++14" when I met the error. Is there any other information I can provide that will help solve this problem?

**Comment by daviddamilola** [comment]
Hello, thanks for you work. Pending the time this is released, what are the steps if i want to use react-native-reanimated from the main branch.

i tried to add it by doing this in package.json
```
"react-native-reanimated": "https://github.com/software-mansion/react-native-reanimated.git#main",
```

which works when building locally, but when doing a  release build it breaks with this error
```
▸ Linking mobile

❌  ld: symbol(s) not found for architecture arm64

❌  clang: error: linker command failed with exit code 1 (use -v to see invocation)

```

**Comment by altaywtf** [comment]
@daviddamilola I'm in a similar situation. 😅 

you may give nightly builds a try: 
- https://www.npmjs.com/package/react-native-reanimated?activeTab=versions 
- https://www.npmjs.com/package/react-native-reanimated/v/3.6.0-nightly-20231114-0837247d4

**Comment by daviddamilola** [comment]
Thanks @altaywtf  , will try it out.

**Comment by UNIDY2002** [comment]
> when doing a release build it breaks with this error

+1

Using the nightly does not seem to solve the problem.

See: https://github.com/thu-info-community/thu-info-app/actions/runs/6891922352/job/18748125712#step:11:2696

Not sure whether anything is wrong on my side.

**Comment by daviddamilola** [comment]
> Thanks @altaywtf , will try it out.

Hello, i can confirm the error remains for v0.73.0-rc.4 but works for v0.73.0-rc.1

**Comment by UNIDY2002** [comment]
> ... when doing a release build it breaks with this error
> 
> ```
> ▸ Linking mobile
> 
> ❌  ld: symbol(s) not found for architecture arm64
> 
> ❌  clang: error: linker command failed with exit code 1 (use -v to see invocation)
> ```

It seems that this issue is fixed by https://github.com/software-mansion/react-native-reanimated/commit/bf7df797139650c3ee9f5891ae5117b8503a9d1d. I've tried on the latest nightly and the build error no longer shows up.

**Comment by mohan-sprinklr** [comment]
> I met below error in RN0.73-rc.2. is there any change with handling hermes? if I comment out enableDebuggging and disableDebugging, build successfully completed.
> 
> ```
> Undefined symbols for architecture x86_64:
>   "facebook::hermes::inspector_modern::chrome::enableDebugging(std::__1::unique_ptr<facebook::hermes::inspector_modern::RuntimeAdapter, std::__1::default_delete<facebook::hermes::inspector_modern::RuntimeAdapter>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)", referenced from:
>       reanimated::ReanimatedHermesRuntime::ReanimatedHermesRuntime(std::__1::unique_ptr<facebook::hermes::HermesRuntime, std::__1::default_delete<facebook::hermes::HermesRuntime>>, std::__1::shared_ptr<facebook::react::MessageQueueThread> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>) in libRNReanimated.a(ReanimatedHermesRuntime.o)
>   "facebook::hermes::inspector_modern::chrome::disableDebugging(int)", referenced from:
>       reanimated::ReanimatedHermesRuntime::~ReanimatedHermesRuntime() in libRNReanimated.a(ReanimatedHermesRuntime.o)
> ld: symbol(s) not found for architecture x86_64
> ```

Did you find the solution for this? 
i am trying to upgrade react-native version from 0.72.6 to 0.73.6 and I have updated the version of react-native-reanimated to its latest version in which I am getting same issue.

**Comment by UNIDY2002** [comment]
@mohan-sprinklr IMHO, this should have been fixed in the latest release of react-native-reanimated. Would you mind confirm whether clearing build cache, re-running pod install and re-building the APP helps?

**Comment by mohan-sprinklr** [comment]
> @mohan-sprinklr IMHO, this should have been fixed in the latest release of react-native-reanimated. Would you mind confirm whether clearing build cache, re-running pod install and re-building the APP helps?

I am still getting the issue in react-native-reanimated latest version i.e 3.8.0
For now I have added a quickfix to manually disable HERMES_DEBUGGER in node_moduels/react-native-reanimated/RNReanimated.podspec file 
image

I cleared pod cache --> deleted Pods folder --> reinstalled pods --> cleared build folder inside xcode --> manually deleted xcode cache inside /Users/<username>/Library/Developer/Xcode
Nothing worked for me.

**Comment by ydvnishant001** [comment]
I think commenting the enableDebugging and disableDebugging option in "node_modules>react-native-reanimated>Common>cpp>ReanimatedRuntime>ReanimatedHermesRuntime.cpp" is better. SInce React Native has deprecated Remote JavaScript Debugging. 

I can no longer see the option in Android dev-menu to remote debug. And the above methods are shown in dev-menu for ios earlier but now are deprecated but still being used by reanimated(I assume). So commenting them makes the build run successfully.

also @mohan-sprinklr , disabling hermes in debug did not fix the issue for me.

If anyone wants to use remote debugging they can follow this official guide - https://reactnative.dev/docs/next/other-debugging-methods. Which works on both ios and android.

**Pull request merged**